pymeasure/instruments/siglenttechnologies/siglent_sds1072cml.py

```diff
@@ -0,0 +1,400 @@
+        (str, {NORM, AUTO, SINGLE,STOP})
+        - "coupling":  (str,{AC,DC}) Coupling to the trigger channel
+
+        Returns a flag indicating if all specified entries were correctly set on the oscilloscope
+        and updates the interal trigger configuration
+        """
+        triggerConfDict = self.get_triggerConfig()
+        # self.triggerConfDict=triggerConfDict
+        setProcesses = {
+            "setup": lambda dictIn: (
+                dictIn.get("type"),
+                dictIn.get("hold_type"),
+                dictIn.get("hold_value1"),
+            ),
+            "level": lambda dictIn: "%.2eV" % dictIn.get("level"),
+            "coupling": lambda dictIn: dictIn.get("coupling"),
+            "slope": lambda dictIn: dictIn.get("slope"),
+            "mode": lambda dictIn: dictIn.get("mode"),
+        }
+        setCommands = {
+            "setup": "TRSE %s,SR,{ch},HT,%s,HV,%s",
+            "level": "{ch}:TRLV %s",
+            "coupling": "{ch}:TRCP %s",
+            "slope": "{ch}:TRSL %s",
+            "mode": "TRMD %s",
+        }
+        setValues = {  # For a given change in conf dict,find the relevant command to be called
+            "setup": ["source", "type", "hold_type", "hold_value1"],
+            "level": ["level"],
+            "coupling": ["coupling"],
+            "slope": ["slope"],
+            "mode": ["mode"],
+        }
+        if kwargs.get("source") is not None:
+            self.id = kwargs["source"]
+        changedValues = [key for key in kwargs if triggerConfDict[key] != kwargs[key]]
+        processToChange = [
+            key for key in setValues if any([value in changedValues for value in setValues[key]])
+        ]
+        for changedKey in changedValues:
+            triggerConfDict[changedKey] = kwargs[changedKey]
+        for processKey in processToChange:
+            self.write(
+                setCommands[processKey] % setProcesses[processKey](triggerConfDict),
+            )
+        self.triggerConfDict = self.get_triggerConfig()
+        statusFlag = self.triggerConfDict == triggerConfDict
+        return statusFlag
+
+
+class SDS1072CML(SCPIMixin, Instrument):
+    """
+    ==============================================
+    Represents the SIGLENT SDS1072CML Oscilloscope
+    ==============================================
+    """
+
+    def __init__(self, adapter, name="Siglent SDS1072CML Oscilloscope", **kwargs):
+        super().__init__(adapter, name, **kwargs)
+
+    channel_1 = Instrument.ChannelCreator(VoltageChannel, "1")
+    channel_2 = Instrument.ChannelCreator(VoltageChannel, "2")
+    trigger = Instrument.ChannelCreator(TriggerChannel, "")
+
+    timeDiv = Instrument.control(
+        ":TDIV?",
+        ":TDIV %s",
+        "Set the time division to the closest possible value,rounding downwards.",
+        validator=truncated_range,
+        values=[5e-9, 50],
+        set_process=lambda v: "%.2eS" % v,
```

Comment:
You can use that set_process logic already in the set_command:
```suggestion
        ":TDIV %.2eS",
        "Set the time division to the closest possible value,rounding downwards.",
        validator=truncated_range,
        values=[5e-9, 50],
```